rust/lance/src/dataset/write.rs

```diff
@@ -644,6 +647,111 @@ async fn resolve_commit_handler(
     }
 }
 
+/// Create an iterator of record batch streams from the given source.
+///
+/// If `enable_retries` is true, the source will be collected, and the
+/// resulting iterator will be an infinite iterator that returns the same
+/// batches over and over again. Otherwise, the source will be yielded by a
+/// one-item iterator.
+///
+/// This is used to support retries on write operations.
```

Comment:
I think this comment is a little confusing.  It doesn't mention spilling at all which I think is a significant detail for anyone learning what this method does.